### ISSUE TITLE
Add a UI option for PyTorch float32 matmul precision (torch.set_float32_matmul_precision) to control TF32 speed/accuracy

### ISSUE THREAD

**Issue by robustini** [ace-step/ACE-Step-1.5#480]
Please add a user-configurable setting for PyTorch float32 matmul precision (highest/high/medium) and apply it early at startup for both inference and training via torch.set_float32_matmul_precision(...). A UI control would be great, but even a simple startup variable (e.g. env/config loaded at launch) is enough. In the current original code there’s no way to enable this without patching files manually. The default should stay highest to preserve current behavior, while high/medium should be opt-in for users who want the TF32 performance trade-off on Ampere+ GPUs.

**Comment by DarkAlchy** [comment]
I didn't know what you meant until I saw this myself: Using bfloat16 Automatic Mixed Precision (AMP)
You are using a CUDA device ('NVIDIA GeForce RTX 4090') that has Tensor Cores. To properly utilize them, you should set `torch.set_float32_matmul_precision('medium' | 'high')` which will trade-off precision for performance. For more details, read https://pytorch.org/docs/stable/generated/torch.set_float32_matmul_precision.html#torch.set_float32_matmul_precision

**Comment by robustini** [comment]
@ChuxiJ why PR closed?

**Comment by ChuxiJ** [comment]
@robustini There’s a conflict. See if you can submit a test PR yourself.

**Comment by robustini** [comment]
> [@robustini](https://github.com/robustini) There’s a conflict. See if you can submit a test PR yourself.

Yes i see, because Copilot is not 100% reliable.
